src/app/jscaip/MoveCoord.ts

```diff
@@ -10,3 +11,27 @@ export abstract class MoveCoord extends Move {
         this.coord = new Coord(x, y);
     }
 }
+
+export class MoveCoordEncoder {
+    public static getEncoder<T extends MoveCoord>(width: number, height: number,
+                                                  construct: (coord: Coord) => T): NumberEncoder<T> {
```

Comment:
generateMove for uniformity, ou renommage de generateMove en "construct" au choix.

Reply:
done